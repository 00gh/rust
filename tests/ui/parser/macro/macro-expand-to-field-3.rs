#![no_main]

macro_rules! field {
    ($name:ident:$type:ty) => {
        $name:$type
    };
}

union EnumVariantField { //~ NOTE while parsing this union
    A: u32,
    field!(oopsies:()), //~ NOTE macros cannot expand to union fields
    //~^ ERROR expected `:`, found `!`
    //~^^ ERROR expected `,`, or `}`, found `(`
    //~^^^ NOTE expected `:`
}
