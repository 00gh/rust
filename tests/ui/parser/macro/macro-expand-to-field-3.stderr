error: expected `:`, found `!`
  --> $DIR/macro-expand-to-field-3.rs:11:10
   |
LL |     field!(oopsies:()),
   |          ^ expected `:`
   |
   = note: macros cannot expand to union fields

error: expected `,`, or `}`, found `(`
  --> $DIR/macro-expand-to-field-3.rs:11:11
   |
LL | union EnumVariantField {
   |       ---------------- while parsing this union
LL |     A: u32,
LL |     field!(oopsies:()),
   |           ^

error: aborting due to 2 previous errors

